From: Cole Robinson <crobinso@redhat.com>
Date: Wed, 8 Aug 2018 14:27:27 -0400
Subject: [PATCH virt-manager] create: Support UEFI installs for armv7l

Will be supported with Fedora 29

https://fedoraproject.org/wiki/Changes/uEFIforARMv7

We should probably limit this by selected OS, but that's tricky
with the wizard as it is, so lets see if anyone cares

(cherry picked from commit cab540792f4fc5ad0270d68d41b54a8fbeea5a7b)
---
 virtManager/create.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/virtManager/create.py b/virtManager/create.py
index 5afe0afe..f0f41ffb 100644
--- a/virtManager/create.py
+++ b/virtManager/create.py
@@ -472,17 +472,17 @@ class vmmCreate(vmmGObjectUI):
         installable_arch = (self._capsinfo.arch in
             ["i686", "x86_64", "ppc64", "ppc64le", "s390x"])
 
-        if self._capsinfo.arch == "aarch64":
+        if self._capsinfo.arch in ["aarch64", "armv7l"]:
             try:
                 guest = self.conn.caps.build_virtinst_guest(self._capsinfo)
                 guest.set_uefi_default()
                 installable_arch = True
-                logging.debug("UEFI found for aarch64, setting it as default.")
+                logging.debug("UEFI found, setting it as default.")
             except Exception as e:
                 installable_arch = False
-                logging.debug("Error checking for aarch64 UEFI default",
+                logging.debug("Error checking for UEFI default",
                     exc_info=True)
-                msg = _("Failed to setup UEFI for AArch64: %s\n"
+                msg = _("Failed to setup UEFI: %s\n"
                         "Install options are limited.") % e
                 self._show_arch_warning(msg)
 
@@ -1864,7 +1864,7 @@ class vmmCreate(vmmGObjectUI):
             self.err.val_err(_("Error setting OS information."), str(e))
             return None
 
-        if guest.os.is_arm64():
+        if guest.os.is_arm64() or guest.os.is_arm():
             try:
                 guest.set_uefi_default()
             except Exception:
